src/Nethermind/Nethermind.Blockchain/Synchronization/FastSync/NodeDataDownloader.cs

```diff
@@ -22,125 +22,69 @@
-                    await getNodeDataTask.ContinueWith(
-                        t =>
+                var hashes = batch.RequestedNodes.Select(r => r.Hash).ToArray();
+                Task<byte[][]> getNodeDataTask = peer.GetNodeData(hashes, token);
+                await getNodeDataTask.ContinueWith(
+                    t =>
+                    {
+                        if (t.IsCompletedSuccessfully)
                         {
-                            if (t.IsCompletedSuccessfully)
-                            {
-                                batch.Responses = getNodeDataTask.Result;
-                            }
+                            batch.Responses = getNodeDataTask.Result;
                         }
-                    );
-                }
-                else
-                {
-                    // _logger.Info("No alloc punishment enabled.");
-                    _noAllocInARow++;
-                    if (_noAllocInARow > 10)
-                    {
-                        _noAllocPunishment = true;
-                        if(_logger.IsInfo) _logger.Info("Adjusting useful peers");
-                        _noAllocInARow = 0;
                     }
-                }
-
-                (NodeDataHandlerResult Result, int NodesConsumed) result = (NodeDataHandlerResult.InvalidFormat, 0);
-                try
-                {
-                    result = _nodeDataFeed.HandleResponse(batch);
-                    if (result.Result == NodeDataHandlerResult.BadQuality)
-                    {
-                        _syncPeerPool.ReportBadPeer(batch.AssignedPeer);
-                    }
-                }
-                catch (Exception e)
-                {
-                    if (_logger.IsError) _logger.Error($"Error when handling response", e);
-                }
-
-                Interlocked.Add(ref _consumedNodesCount, result.NodesConsumed);
-                if (result.NodesConsumed == 0 && peer != null)
-                {
-                    _syncPeerPool.ReportNoSyncProgress(nodeSyncAllocation);
-                }
+                );
             }
-            finally
+
+            (NodeDataHandlerResult Result, int NodesConsumed) result = (NodeDataHandlerResult.InvalidFormat, 0);
+            try
             {
-                if (nodeSyncAllocation != null)
+                result = _feed.HandleResponse(batch);
+                if (result.Result == NodeDataHandlerResult.BadQuality)
                 {
-                    _syncPeerPool.Free(nodeSyncAllocation);
+                    _syncPeerPool.ReportBadPeer(batch.AssignedPeer);
                 }
             }
-        }
-
-        private async Task UpdateParallelism()
-        {
-            int newUsefulPeerCount = Math.Max(0, _syncPeerPool.UsefulPeerCount - (_noAllocPunishment ? 1 : 0));
-            int difference = newUsefulPeerCount - _lastUsefulPeerCount;
-            if (difference == 0)
+            catch (Exception e)
```

Comment:
DO we really want to swallow all exceptions here?

Reply:
yeah - this is unrecoverble error - it will display a lot of issues and then user is exepcted to close the app

Reply:
But maybe we can narrow the exception types? If not its fine